extensions/amp-ad-network-base/0.1/amp-ad-network-base.js

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdUIHandler} from '../../amp-ad/0.1/amp-ad-ui'; // eslint-disable-line no-unused-vars
+import {
+  LayoutInfoDef,
+  getAmpAdMetadata,
+  sendXhrRequest,
+} from '../../amp-a4a/0.1/a4a-utils';
+import {
+  RendererDef,
+  RendererInputDef,
+  ValidatorDef, // eslint-disable-line no-unused-vars
+  ValidatorOutputDef,
+  ValidatorResultType, // eslint-disable-line no-unused-vars
```

Comment:
can we have those type def in a separate file?

Reply:
Not in amp-render, you mean? I was thinking that might be a good idea to declutter that file.

Reply:
We should have interfaces in a separate file from the implementation, so that importing an interface will not get all the impl code compiled into the extension (though Closure compiler might be smart with that, but it's just a good practice).